homeassistant/components/husqvarna_automower/lawn_mower.py

```diff
@@ -0,0 +1,124 @@
+"""Creates a vacuum entity for the mower."""
+import logging
+
+from aiohttp import ClientResponseError
+
+from homeassistant.components.lawn_mower import (
+    LawnMowerActivity,
+    LawnMowerEntity,
+    LawnMowerEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, ERROR_ACTIVITIES
+from .entity import AutomowerEntity
+
+SUPPORT_STATE_SERVICES = (
+    LawnMowerEntityFeature.DOCK
+    | LawnMowerEntityFeature.PAUSE
+    | LawnMowerEntityFeature.START_MOWING
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up vacuum platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        HusqvarnaAutomowerEntity(coordinator, idx)
+        for idx, ent in enumerate(coordinator.session.data["data"])
+    )
+    entity_platform.current_platform.get()
+
+
+class HusqvarnaAutomowerEntity(LawnMowerEntity, AutomowerEntity):
+    """Defining each mower Entity."""
+
+    _attr_name: str | None = None
+    _attr_supported_features = SUPPORT_STATE_SERVICES
+
+    def __init__(self, session, idx) -> None:
+        """Set up HusqvarnaAutomowerEntity."""
+        super().__init__(session, idx)
+        self._attr_unique_id = self.coordinator.session.data["data"][self.idx]["id"]
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available."""
+        available = self.get_mower_attributes()["metadata"]["connected"]
```

Comment:
I'd recommend creating dataclasses in the library from the parsed json data to make it easier for the library user to use the data returned from the api.